src/scp.c

```diff
@@ -860,9 +870,18 @@ scp_send(LIBSSH2_SESSION * session, const char *path, int mode,
                  "scp -%st ", (mtime || atime)?"p":"");
 
         cmd_len = strlen((char *)session->scpSend_command);
-        cmd_len += shell_quotearg(path,
-                                  &session->scpSend_command[cmd_len],
-                                  session->scpSend_command_len - cmd_len);
+
+        if(!session->flag.quote_paths) {
+            path_len = strlen(path);
+            memcpy(&session->scpSend_command[cmd_len], path, path_len); /* no NUL-termination neeed, so memcpy will do */
```

Comment:
Comment needs to be moved to a separate line to pass `checksrc`.

Reply:
Done, pushed a second ago.